macaroons/store_test.go

```diff
@@ -169,6 +170,37 @@ func TestStoreGenerateNewRootKey(t *testing.T) {
 	require.NotEqual(t, oldRootKey, newRootKey)
 }
 
+// TestStoreSetRootKey tests that a root key can be set to a specified value.
+func TestStoreSetRootKey(t *testing.T) {
+	_, cleanup, store := newTestStore(t)
+	defer cleanup()
+
+	// Create a new key
+	rootKey := make([]byte, 32)
+	_, err := rand.Read(rootKey)
+	require.NoError(t, err)
+
+	// The store must be unlocked to set the root key.
+	err = store.SetRootKey(rootKey)
+	require.Equal(t, macaroons.ErrStoreLocked, err)
+
+	// Unlock the store and read the current key.
```

Comment:
We don't actually read the current key. Maybe we should, just to make sure it's not yet set to the one we're about to set?

Reply:
I just removed this comment for now, personally I don't see why lnd should prevent setting the root key to its current value. But if you or someone feels strongly about not allowing that I can add that logic in.